src/snowflake/snowpark/functions.py

```diff
@@ -3496,6 +3496,139 @@ def array_intersection(array1: ColumnOrName, array2: ColumnOrName) -> Column:
     return builtin("array_intersection")(a1, a2)
 
 
+def array_except(
+    source_array: ColumnOrName, array_of_elements_to_exclude: ColumnOrName
+) -> Column:
+    """Returns a new ARRAY that contains the elements from one input ARRAY that are not in another input ARRAY.
+
+    The function is NULL-safe, meaning it treats NULLs as known values for comparing equality.
+
+    ARRAY_EXCEPT compares arrays by using multi-set semantics (sometimes called “bag semantics”). If source_array
+    includes multiple copies of a value, the function only removes the number of copies of that value that are specified
+    in array_of_elements_to_exclude.
+
+    For example, if source_array contains 5 elements with the value 'A' and array_of_elements_to_exclude contains 2
+    elements with the value 'A', the returned array contains 3 elements with the value 'A'.
+
+    Args:
+        source_array: An array that contains elements to be included in the new ARRAY.
+        array_of_elements_to_exclude: An array that contains elements to be excluded from the new ARRAY.
+
+    Example::
+        >>> from snowflake.snowpark import Row
+        >>> df = session.create_dataframe([Row(["A", "B"], ["B", "C"])], schema=["a", "b"])
```

Comment:
That's a lot of a's and b's. Could the schema be `col1` and `col2` to better separate the meanings?

Reply:
I'm following other tests, but I could change the name to be more descriptive :)